### PR TITLE
Add case for copy storage migration with no disk

### PR DESCRIPTION
**Pull request by cliping**
VIRT-294655 - Migrate vm with copy storage - vm has no disk

Signed-off-by: lcheng <lcheng@redhat.com>

### CODE REVIEW

**Comment by cliping** [comment]
Test result:

#avocado run --vt-type libvirt --vt-machine-type q35 virsh.live_migration.copy_storage_migration_with_no_disk.copy_storage_all.p2p_live.without_postcopy --vt-connect-uri qemu:///system
JOB ID     : efe99e8ec54b28a05281d9d41e21e1615a6bf243
JOB LOG    : /root/avocado/job-results/job-2022-04-17T21.59-efe99e8/job.log
 (1/1) type_specific.io-github-autotest-libvirt.virsh.live_migration.copy_storage_migration_with_no_disk.copy_storage_all.p2p_live.without_postcopy: PASS (56.23 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 56.89 s

#avocado run --vt-type libvirt --vt-machine-type q35 virsh.live_migration.copy_storage_migration_with_no_disk.copy_storage_inc.p2p_live.without_postcopy --vt-connect-uri qemu:///system
JOB ID     : 6741138313afcaad295d1558774a84f93dcaf9f5
JOB LOG    : /root/avocado/job-results/job-2022-04-17T22.02-6741138/job.log
 (1/1) type_specific.io-github-autotest-libvirt.virsh.live_migration.copy_storage_migration_with_no_disk.copy_storage_inc.p2p_live.without_postcopy: PASS (56.47 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 57.20 s

File: libvirt/tests/src/migration/live_migration.py [hunk 1]
```diff
@@ -293,12 +303,16 @@ def run(test, params, env):
             remote_old.run_remote_cmd(cmd, params, runner_on_target, ignore_status=False)
 
         if not vm.is_alive():
-            vm.start()
+            if copy_storage_migration_with_no_disk:
+                virsh.start(vm_name, debug=True, ignore_status=False)
+                time.sleep(10)
```
Comment (chunfuwen) [inline libvirt/tests/src/migration/live_migration.py:308]: Could we use vm.wait_for_login().close() rather hard code sleep timeout value?

File: libvirt/tests/cfg/migration/live_migration.cfg [hunk 2]
```diff
@@ -260,3 +260,16 @@
             exec_statistics_cmd = 'yes'
             migrate_speed = 10
             action_during_mig = '[{"func": "execute_statistics_command", "after_event": "iteration: '1'", "func_param": "params"}]'
+        - copy_storage_migration_with_no_disk:
```
Comment (Yingshun) [inline libvirt/tests/cfg/migration/live_migration.cfg:263]: Would you please move it to migrate_storage or create a new file for storage migration?

**Comment by cliping** [comment]
Test result:

# avocado run --vt-type libvirt --vt-machine-type q35 migration.virtual_disk_migration.copy_storage_migration_with_no_disk --vt-connect-uri qemu:///system
JOB ID     : 34ad11ae7a03328322b5a02279c1851edb14e4f4
JOB LOG    : /root/avocado/job-results/job-2022-05-07T04.54-34ad11a/job.log
 (1/2) type_specific.io-github-autotest-libvirt.migration.virtual_disk_migration.copy_storage_migration_with_no_disk.copy_storage_all.p2p_live.with_precopy: PASS (37.96 s)
 (2/2) type_specific.io-github-autotest-libvirt.migration.virtual_disk_migration.copy_storage_migration_with_no_disk.copy_storage_inc.p2p_live.with_precopy: PASS (37.08 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 75.82 s

File: libvirt/tests/cfg/migration/live_migration.cfg (hunk 2 above)
Comment (cliping) [reply libvirt/tests/cfg/migration/live_migration.cfg]: Updated, create a new file for this case. Thanks.

File: libvirt/tests/src/migration/live_migration.py (hunk 1 above)
Comment (cliping) [reply libvirt/tests/src/migration/live_migration.py]: The vm in this case don't have disk. Use vm.wait_for_login() will failed.

File: libvirt/tests/cfg/migration/virtual_disk_migration.cfg [hunk 3]
```diff
@@ -0,0 +1,33 @@
+- migration.virtual_disk_migration:
```
Comment (chunfuwen) [inline libvirt/tests/cfg/migration/virtual_disk_migration.cfg:1]: libvirt/tests/cfg/migration/migrate_storage.cfg, this already cover part of copy_storage_inc, are there some overlapping here?

Comment (cliping) [reply libvirt/tests/cfg/migration/virtual_disk_migration.cfg]: Add this case for copy storage migration modular design.

**Review by Yingshun — APPROVED** [review]

**Review by chunfuwen — APPROVED** [review]
LGTM

**Pull request merged**